### PR TITLE
Fix IDF tool installation

### PR DESCRIPTION
**Pull request by mikee47**
This PR fixes tool installation for IDF < 5.2.

Overlooked that wildcard tool spec. only supported for IDF 5.2 installer.
Fixed by providing explicit list of tools for each version.

**Improve cache-clean action**

Not very intuitive so added four levels of cleaning:

- pull-requests: Any items created by pull requests, excludes anything in develop branch
- ccache: All ccache items
- idf-tools: All IDF tools
- ccache+idf: All ccache and IDF tool caches.

Removed the `cache-rebuild` action since we can just do a cache clean then manually trigger a rebuild of the develop branch to accomplish a similar thing.

**Update `clean-tools.py` script**

Didn't clean some tools.

### CODE REVIEW

**Comment by mikee47** [comment]
I cleaned the appropriate IDF caches before creating this PR so will actually run the install scripts to verify.

File: Sming/Arch/Esp32/Tools/install.sh [hunk 1]
```diff
@@ -61,22 +61,24 @@ rm -f "$IDF_PATH"
...
-fi
-python3 "$IDF_PATH/tools/idf_tools.py" --non-interactive install-python-env
+# Install IDF tools and packages (unless CI has restored from cache)
+if [ -n "$CI_BUILD_DIR" ] && [ -d "$IDF_TOOLS_PATH/tools" ]; then
+    printf "\n\n** Skipping IDF tools installation: '%s/tools' exists\n\n" "$IDF_TOOLS_PATH"
+else
+    # Be specific about which tools we want to install for each IDF version
+    IDF_TOOL_PACKAGES=$(tr '\n' ' ' < "$SMING_HOME/Arch/Esp32/Tools/idf_tools-${INSTALL_IDF_VER}.lst")
+    echo "Install: $IDF_TOOL_PACKAGES"
+    python3 "$IDF_PATH/tools/idf_tools.py" --non-interactive install $IDF_TOOL_PACKAGES
```
Comment (mikee47) [inline Sming/Arch/Esp32/Tools/install.sh:71]: Codacy persistently complains when variables such as `$IDF_TOOL_PACKETS` are used without quotes. I'm not clear on the best way to shut it up, but this usage is intended.

Comment (slaff) [reply Sming/Arch/Esp32/Tools/install.sh:71]: > clear on the best way to shut it up, but this usage is intended.

I will do it. Basically you need to click on "Ignore issue" for this file.

Comment (slaff) [reply Sming/Arch/Esp32/Tools/install.sh:71]: Done.

**Comment by slaff** [comment]
@mikee47 do you have something else to add to this PR?

**Comment by mikee47** [comment]
@slaff Nope, all done until something else breaks :-)

**Comment by mikee47** [comment]
I've wiped all the caches so this gets a proper run when it's merged

**Pull request merged**